common/src/chain/transaction/signature/inputsig/standard_signature.rs

```diff
@@ -67,44 +69,14 @@ impl StandardInputSignature {
         Ok(decoded_sig)
     }
 
-    pub fn verify_signature(
-        &self,
-        chain_config: &ChainConfig,
-        outpoint_destination: &Destination,
-        sighash: &H256,
-    ) -> Result<(), DestinationSigError> {
-        match outpoint_destination {
-            Destination::PublicKeyHash(addr) => {
-                let sig_components = AuthorizedPublicKeyHashSpend::from_data(&self.raw_signature)?;
-                verify_address_spending(addr, &sig_components, sighash)?
-            }
-            Destination::PublicKey(pubkey) => {
-                let sig_components = AuthorizedPublicKeySpend::from_data(&self.raw_signature)?;
-                verify_public_key_spending(pubkey, &sig_components, sighash)?
-            }
-            Destination::ScriptHash(_) => return Err(DestinationSigError::Unsupported),
-            Destination::AnyoneCanSpend => {
-                // AnyoneCanSpend must use InputWitness::NoSignature, so this is unreachable
-                return Err(
-                    DestinationSigError::AttemptedToVerifyStandardSignatureForAnyoneCanSpend,
-                );
-            }
-            Destination::ClassicMultisig(h) => {
-                let sig_components =
-                    AuthorizedClassicalMultisigSpend::from_data(&self.raw_signature)?;
-                verify_classical_multisig_spending(chain_config, h, &sig_components, sighash)?
-            }
-        }
-        Ok(())
-    }
-
     pub fn produce_uniparty_signature_for_input<T: Signable>(
         private_key: &crypto::key::PrivateKey,
         sighash_type: SigHashType,
         outpoint_destination: Destination,
         tx: &T,
         inputs_utxos: &[Option<&TxOutput>],
         input_num: usize,
+        htlc_secret: Option<HtlcSecret>,
```

Comment:
I suggest creating another function that does the htlc stuff and that calls this one. This Option stuff is a bad idea.